orchagent/neighorch.cpp

```diff
@@ -1099,20 +1109,131 @@ void NeighOrch::doVoqSystemNeighTask(Consumer &consumer)
-    //For "port" type inband, the inband reachability info syncing can be done through static
-    //configureation or CHASSIS_APP_DB sync (this function)
+    //Add neighbor record in SAI without adding host route for local inband to avoid route
+    //looping for packets destined to the Inband interface if the Inband is port type
+
+    if(gIntfsOrch->isRemoteSystemPortIntf(alias))
+    {
+        //Remote Inband interface. Skip
+        return true;
+    }
+
+    sai_status_t status;
+    MacAddress inband_mac = gMacAddress;
+
+    sai_object_id_t rif_id = gIntfsOrch->getRouterIntfsId(alias);
+    if (rif_id == SAI_NULL_OBJECT_ID)
+    {
+        SWSS_LOG_INFO("Failed to get rif_id for %s", alias.c_str());
+        return false;
+    }
+
+    //Make the object key
+    sai_neighbor_entry_t neighbor_entry;
+    neighbor_entry.rif_id = rif_id;
+    neighbor_entry.switch_id = gSwitchId;
+    copy(neighbor_entry.ip_address, ip_address);
+
+    vector<sai_attribute_t> neighbor_attrs;
+    sai_attribute_t attr;
+    attr.id = SAI_NEIGHBOR_ENTRY_ATTR_DST_MAC_ADDRESS;
+    memcpy(attr.value.mac, inband_mac.getMac(), 6);
+    neighbor_attrs.push_back(attr);
+
+    if(!addVoqEncapIndex(alias, ip_address, neighbor_attrs))
+    {
+        return false;
+    }
+
+    //No host route for neighbor of the Inband IP address
+    attr.id = SAI_NEIGHBOR_ENTRY_ATTR_NO_HOST_ROUTE;
+    attr.value.booldata = true;
+    neighbor_attrs.push_back(attr);
 
-    //For "vlan" type inband, the inband reachability info syncinng can be ARP learning of other
-    //asics inband or static configuration or through CHASSIS_APP_DB sync (this function)
+    status = sai_neighbor_api->create_neighbor_entry(&neighbor_entry, static_cast<uint32_t>(neighbor_attrs.size()), neighbor_attrs.data());
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        if (status == SAI_STATUS_ITEM_ALREADY_EXISTS)
+        {
+            SWSS_LOG_ERROR("Entry exists: neighbor %s on %s, rv:%d", inband_mac.to_string().c_str(), alias.c_str(), status);
+            return true;
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Failed to create neighbor %s on %s, rv:%d", inband_mac.to_string().c_str(), alias.c_str(), status);
+            return false;
+        }
+    }
 
-    //May implement inband rechability info syncing through CHASSIS_APP_DB sync here
+    SWSS_LOG_NOTICE("Created inband neighbor %s on %s", inband_mac.to_string().c_str(), alias.c_str());
+
+    gIntfsOrch->increaseRouterIntfsRefCount(alias);
+
+    if (neighbor_entry.ip_address.addr_family == SAI_IP_ADDR_FAMILY_IPV4)
+    {
+        gCrmOrch->incCrmResUsedCounter(CrmResourceType::CRM_IPV4_NEIGHBOR);
+    }
+    else
+    {
+        gCrmOrch->incCrmResUsedCounter(CrmResourceType::CRM_IPV6_NEIGHBOR);
+    }
+
+    //Sync the neighbor to add to the CHASSIS_APP_DB
+    voqSyncAddNeigh(alias, ip_address, inband_mac, neighbor_entry);
 
     return true;
 }
 
 bool NeighOrch::delInbandNeighbor(string alias, IpAddress ip_address)
```

Comment:
When will the local inband neigh be deleted? It should never happen, right?

Reply:
Yes. So far I do not have a compelling situation when we'll delete Inband neigh. Right now this API is not used anywhere since we do not delete the Inband interface once configured. We also do not support changing this once configured. May be in the future we may have a situation to dynamically change inband interface or ip address, when we'll use this.